### PR TITLE
added task memory game

### PR DESCRIPTION
**Pull request by chris-voitova**
### Memory - Pair Game

[Demo](https://chris-voitova.github.io/memory-pair-game/) |
[Code base](https://github.com/chris-voitova/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by zonzujiro** [comment]
image

Oh-oh! 🙈 You should have 2 cards maximum :)

**Comment by chris-voitova** [comment]
> image
> 
> Oh-oh! 🙈 You should have 2 cards maximum :)

fixed 🥳

File: submissions/chris-voitova/memory-game/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+function goToInitState(cards) {
+  setTimeout(() => {
+    shuffleCards(cards);
+    createCardsFromArr(cards);
+    cardsAmount = cards.length;
+  }, 2000);
+}
+
+function checkTwoCards(field) {
```
Comment (MaxKalashnyk) [inline submissions/chris-voitova/memory-game/index.js:48]: Do we really need to pass an argument to this function?

File: submissions/chris-voitova/memory-game/index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+              ),
+            1000
+          );
+        checkedCards = [];
+      }
+    });
+  }
+}
+
+function toggleClassName(element, classNames) {
```
Comment (MaxKalashnyk) [inline submissions/chris-voitova/memory-game/index.js:85]: 👍🏻

File: submissions/chris-voitova/memory-game/index.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  element.classList.remove(...classNames);
+  element.classList.add(...classNames);
+}
+
+shuffleCards(cards);
+createCardsFromArr(cards);
+
+field.addEventListener("click", function (e) {
```
Comment (MaxKalashnyk) [inline submissions/chris-voitova/memory-game/index.js:93]: You can descture `event` here

File: submissions/chris-voitova/memory-game/index.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    createCardsFromArr(cards);
+    cardsAmount = cards.length;
+  }, 2000);
+}
+
+function checkTwoCards(field) {
+  let checkedCards = [];
+  cardsInField = Array.from(field.children);
+
+  cardsInField.forEach((item) => {
```
Comment (MaxKalashnyk) [inline submissions/chris-voitova/memory-game/index.js:52]: Maybe using `Array.prototype.filter` will be better to use here

File: submissions/chris-voitova/memory-game/index.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    checkedCards.reduce((first, curr, index, arr) => {
+      if (first.dataset.name === curr.dataset.name) {
+        cardsAmount -= 2;
+        setTimeout(
+          () =>
+            arr.forEach((item) => {
+              item.classList.remove("active");
+              item.classList.add("hidden");
+            }, (isBlockedField = false)),
+          1000
```
Comment (MaxKalashnyk) [inline submissions/chris-voitova/memory-game/index.js:66]: Maybe it's better to move this value to some constant

File: submissions/chris-voitova/memory-game/index.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        checkedCards = [];
+      } else {
+        arr.forEach((item) => item.classList.remove("active")),
+          setTimeout(
+            () =>
+              arr.forEach(
+                (item) => item.classList.remove("flipped"),
+                (isBlockedField = false)
+              ),
+            1000
```
Comment (MaxKalashnyk) [inline submissions/chris-voitova/memory-game/index.js:77]: the same as in the previous comment

File: submissions/chris-voitova/memory-game/index.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+field.addEventListener("click", function (e) {
+  const target = e.target;
+  if (target.classList.contains("card") && !isBlockedField) {
+    toggleClassName(target, ["active", "flipped"]);
+    checkTwoCards(field);
+  }
+  if (cardsAmount === 0) {
+    setTimeout(() => {
+      alert("u are win!");
+    }, 2000);
```
Comment (MaxKalashnyk) [inline submissions/chris-voitova/memory-game/index.js:102]: Also, it's better to move it constant

File: submissions/chris-voitova/memory-game/index.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+field.addEventListener("click", function (e) {
+  const target = e.target;
+  if (target.classList.contains("card") && !isBlockedField) {
+    toggleClassName(target, ["active", "flipped"]);
+    checkTwoCards(field);
+  }
+  if (cardsAmount === 0) {
+    setTimeout(() => {
+      alert("u are win!");
```
Comment (MaxKalashnyk) [inline submissions/chris-voitova/memory-game/index.js:101]: Maybe, this string would be better to store in some constant.

**Review by MaxKalashnyk — COMMENTED** [review]
Nice work! I really like it.
Your code is so clean. Respect for that! Look at my minor suggestions, please

**Comment by chris-voitova** [comment]
> Nice work! I really like it.
> Your code is so clean as well. Appreciate it! 👍🏻
> Look at my minor suggestions, please

Thanks!! I made fixes, please check

**Pull request merged**